### PR TITLE
Authenticationcheck

### PR DESCRIPTION
**Pull request by ianlovell**

### CODE REVIEW

File: app/containers/AuthCallback.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React, { Component } from 'react';
```
Comment (csscottlogic) [inline app/containers/AuthCallback.jsx]: Sorry for the noob question, but is this something to hang stuff from, or is it just a stub to get the development going?

File: app/main.js [hunk 2]
```diff
@@ -1,39 +1,50 @@
...
+AuthService.isAuthenticated();
 
 import injectTouchTapEvent from 'react-tap-event-plugin';
 injectTouchTapEvent();
 
 const store = configureStore();
 
 const history = syncHistoryWithStore(browserHistory, store, {
-    selectLocationState: state => (state.get('routing') ? state.get('routing').toJS() : state),
+    selectLocationState: state => (state.get('routing')
```
Comment (csscottlogic) [inline app/main.js]: Presumably ```state.get('routing')```

File: app/main.js [hunk 3]
```diff
@@ -1,39 +1,50 @@
...
-      <Route path="sessions" component={App}>
-        <Route path=":sessionId" component={Session} />
-      </Route>
-      <Route path="calendar" component={App}>
-        <IndexRoute component={Calendar} />
-      </Route>
-      <Redirect from="*" to="/" />
+        <Route component={AuthCallback} />
+        <Route component={App}>
+            <Route path="/dashboard" component={userIsAuthenticated(Dashboard)} />
```
Comment (csscottlogic) [inline app/main.js]: Are we only enforcing authentication to view the dashboard initially? (To get the plumbing in.)

File: app/main.js [hunk 4]
```diff
@@ -1,39 +1,50 @@
...
+import { UserAuthWrapper } from 'redux-auth-wrapper';
 
 import App from './containers/App.jsx';
+import AuthCallback from './containers/AuthCallback.jsx';
 import Dashboard from './containers/Dashboard.jsx';
 import Session from './containers/Session.jsx';
 import Calendar from './containers/Calendar.jsx';
+import AuthService from './services/AuthService.js';
+
+AuthService.isAuthenticated();
```
Comment (csscottlogic) [inline app/main.js]: Currently called because it contains a ```console.log```?

File: app/services/AuthService.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        const settings = {
+            authority: 'http://api.bris.tech:9003',
+            client_id: 'bristechsrm',
+            redirect_uri: 'http://srm.bris.tech/signed-in',
+            post_logout_redirect_uri: 'http://srm.bris.tech',
+            response_type: 'id_token token',
+            scope: 'openid profile api',
+        };
+
+        this.tokenManager = new OidcTokenManager(settings);
```
Comment (csscottlogic) [inline app/services/AuthService.js]: ```OidcTokenManager``` is on ```window```? I'm just wondering why there's no import.

File: app/main.js (hunk 3 above)
Comment (ianlovell) [inline app/main.js]: Yes, will apply to all(?) routes once OIDC Provider is up

File: app/services/AuthService.js (hunk 5 above)
Comment (ianlovell) [inline app/services/AuthService.js]: unfortunately oidc-token-provider is not an npm package/commonjs/umd format, so we can't import without issues (and setting up with webpack with bower). They have re-written the package in ES6, and is due for release in ~2 weeks, so when that's released, we can import and avoid the global.

File: app/main.js (hunk 4 above)
Comment (ianlovell) [inline app/main.js]: Correct, yes just testing out until the callback page is implemented (coming next)

File: app/main.js (hunk 2 above)
Comment (ianlovell) [inline app/main.js]: Is quick, but will factor out as part of the auth callback work.

File: app/containers/AuthCallback.jsx (hunk 1 above)
Comment (ianlovell) [inline app/containers/AuthCallback.jsx]: Just a stub to fill out later

**Comment by ianlovell** [comment]
:shipit: :rocket:

**Pull request merged**